extensions-core/kubernetes-extensions/src/main/java/org/apache/druid/k8s/discovery/DefaultK8sApiClient.java

```diff
@@ -131,7 +131,7 @@ public boolean hasNext() throws SocketTimeoutException
           try {
             while (watch.hasNext()) {
               Watch.Response<V1Pod> item = watch.next();
-              if (item != null && item.type != null) {
+              if (item != null && item.type != null && !item.type.equals(WatchResult.BOOKMARK)) {
```

Comment:
Have you tried setting allowWatchBookmarks to false on line 116

https://github.com/apache/druid/pull/15819/files#diff-93450087400a4498aaf2b84919eb6172819f1df4b14d29589f9e29d456611bbdR116

Reply:
We're actually not totally ignoring the result, we are updating the resourceVersion which helps to reduce load on Kubernetes API so it does make sense to get those. See for details https://kubernetes.io/docs/reference/using-api/api-concepts/#watch-bookmarks

https://github.com/apache/druid/blob/2e46a980245bc1f2694dd21af9425855e78632f9/extensions-core/kubernetes-extensions/src/main/java/org/apache/druid/k8s/discovery/K8sDruidNodeDiscoveryProvider.java#L282

Reply:
ah! thanks for pointing that out